### PR TITLE
refactor: [M3-8653] - Migrate `Toggle` component to `linode/ui` package

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
Migrate `Toggle` component to `linode/ui` package.

## Changes  🔄
- Migrate `Toggle` component to `linode/ui` package
- Migrate `ToggleOff` and `ToggleOn` icon to `linode/ui` package

## Target release date 🗓️
12/10/24

### Verification steps
- All tests should pass
- There should be no regressions observed in Cloud
- `Toggle` story in Storybook should be in good shape

## As an Author, I have considered 🤔
- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

### CODE REVIEW

File: packages/ui/src/components/BetaChip/BetaChip.test.tsx [hunk 1]
```diff
@@ -1,11 +1,10 @@
+import '@testing-library/jest-dom/vitest';
```
Comment (coliu-akamai) [inline packages/ui/src/components/BetaChip/BetaChip.test.tsx:1]: for any `ui` package file changes that aren't the `ToggleOn`/`ToggleOff` icons or the `Toggle` component, these are just eslint/prettier changes that I did while fixing some merge conflicts

File: packages/ui/src/foundations/themes/index.ts [hunk 2]
```diff
@@ -143,25 +143,25 @@ declare module '@mui/material/styles/createTheme' {
...
+      //  ---- Global tokens: theme agnostic ----
+      borderRadius?: BorderRadiusTypes;
       chart?: ChartTypes;
+      color?: ColorTypes;
       content?: ContentTypes;
       elevation?: ElevationTypes;
+      font?: FontTypes;
       interaction?: InteractionTypes;
       radius?: RadiusTypes;
+      spacing?: SpacingTypes;
```
Comment (jaalah-akamai) [inline packages/ui/src/foundations/themes/index.ts:162]: I'm fine with this if the distinction between global and alias isn't clear - however we may want to either remove the code comments or put these back as they were.

**Review by jaalah-akamai — APPROVED** [review]

File: packages/ui/src/foundations/themes/index.ts (hunk 2 above)
Comment (coliu-akamai) [reply packages/ui/src/foundations/themes/index.ts:162]: good point - imo i do like the comments, so will move this specific file back to as it was.  ty!!

File: packages/ui/src/components/Tooltip/Tooltip.stories.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import { Meta, StoryObj } from '@storybook/react';
 import React from 'react';
 
 import { Tooltip } from './Tooltip';
 
+import type { Meta, StoryObj } from '@storybook/react';
 // @todo: modularization - Import from 'ui' package once Typography is migrated.
 import { Typography } from '@mui/material';
```
Comment (pmakode-akamai) [inline packages/ui/src/components/Tooltip/Tooltip.stories.tsx:7]: Now that `Typography` is migrated, we can import it from the `ui` package (note: we have this comment in a few other places in the `ui` package)

**Review by pmakode-akamai — APPROVED** [review]
thanks Connie! Changes LGTM ✅

Toggle in Cloud ✅
Storybook ✅

**Pull request merged**